vertx-config/src/main/java/io/vertx/config/spi/utils/JsonObjectHelper.java

```diff
@@ -113,7 +115,11 @@ private static JsonArray asJsonArray(String s) {
       } catch (Exception e) {
         return null;
       }
+    } else if (!s.startsWith("[") && !s.endsWith("]") && s.contains(",")) {
+      // Allow same comma-separated syntax as for .properties files
+      return asJsonArray("[" + s + "]");
     }
```

Comment:
I added this case for array parsing to be more consistent with the parsing method used by the `PropertiesConfigProcessor`. However, there are still differences, as the latter does not allow the full JSON array syntax (including the square braces) when `hierarchical = true`.

I really think that the `PropertiesConfigProcessor` should also be using `JsonObjectHelper.convert(String)` to parse values. That would make for easier reasoning about the results. Additionally, it would be an option to drop the custom properties parsing logic from `PropertiesConfigProcessor.HierarchicalPropertiesReader` in favor of letting the JDK do the parsing and convert the resulting `Properties` using `JsonObjectHelper.from(...)`. That would be the most consistent option, in my opinion. What do you think, @vietj?

Reply:
I think it's a good idea, can we proceed in this PR ?

Reply:
Yes, I will implement it as proposed

Reply:
Done